### DISCUSSION TITLE
Automatic supplier price update

### DISCUSSION THREAD

**Discussion by dieselburner** [inventree/InvenTree#7226] [General]
As per title - I have checked available plugins and settings, but not sure whether this is possible at the moment. If not - would it be possible to implement automatic periodic (weekly) job (via plug-in?) that would update all the prices in bulk? I do know at least [DigiKey](https://developer.digikey.com/) and [Mouser](https://eu.mouser.com/api-solutions/) exposes API, but the app needs to be registered there.

Sorry for the waste of your time if this functionality already exists - would appreciate if you guide me. Much obliged.

**Comment by dieselburner** [comment]
While I'm still thinking about this - pricing information can be retrieved automatically as well, after component creation, potentially via the same functionality.

**Comment by matmair** [comment]
There have been discussions about this and I even built an MVP but there 2 big problems:
1) You are not supposed to keep the (pricing) info accumulated in databases according to the TOS of both vendors you mentioned
2) I got dinged in the past by vendors for scraping / LCSC for example makes a game out of constantly changing their APIs and adding more crawling disruptors.

We would also not be the first project to get a C&D which is why I more or less stopped efforts in this area.

**Comment by SergeoLacruz** [comment]
I created a plugin for Mouser
https://github.com/SergeoLacruz/inventree-supplier-plugin
This one crawls through your supplier parts and updates the prices. It gets one part a time and runs every few minutes. The calls to the Mouser API are limited per time period. This works but the plugin needs a major redesign, like getting several parts at once. The code needs to be cleaned up, ...
You need to register with Mouser to use their API. You get an access key that the plugin needs. The Mouser API is simple and stable. 

For Digikey the access method is more complicated. This one
https://github.com/SergeoLacruz/inventree-supplier-panel
gets parts and uploads purchase orders to Digikey and Mouser. It does not handle the pricing information because this shall be done by the other plugin. The Digikey API is stable, at least during the last year, but complicated.

Hope that helps.

**Comment by dieselburner** [comment]
Thanks for your inputs!
